src/client/App.jsx

```diff
@@ -1,30 +1,94 @@
-import { useEffect, useState } from 'react';
-import './App.css';
-import MovieForm from './components/MovieForm';
-import UserForm from './components/UserForm';
+import { useEffect, useState } from "react";
+import "./App.css";
+import MovieForm from "./components/MovieForm";
+import UserForm from "./components/UserForm";
 
-const apiUrl = 'http://localhost:4000';
+const apiUrl = "http://localhost:4000";
 
 function App() {
   const [movies, setMovies] = useState([]);
 
   useEffect(() => {
     fetch(`${apiUrl}/movie`)
-      .then(res => res.json())
-      .then(res => setMovies(res.data));
+      .then((res) => res.json())
+      .then((res) => setMovies(res.data));
   }, []);
 
   const handleRegister = async ({ username, password }) => {
-    
+    try {
+      const response = await fetch(`http://localhost:4000/user/register`, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({ username, password }),
+      });
+
+      if (!response.ok) {
+        throw new Error(`HTTP error! status: ${response.status}`);
+      }
+
+      const data = await response.json();
+      console.log("User registered:", data);
+    } catch (error) {
+      console.error("Error during user registration:", error);
+    }
   };
 
   const handleLogin = async ({ username, password }) => {
-    
+    try {
+      const response = await fetch(`http://localhost:4000/user/login`, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({ username, password }),
+      });
+
+      if (!response.ok) {
+        throw new Error(`HTTP error! status: ${response.status}`);
+      }
+      const { token } = await response.json();
+
+      localStorage.setItem("token", token);
+
+      console.log("User logged in:", username, localStorage.getItem("token"));
+    } catch (error) {
+      console.error("Error during user login:", error);
+    }
   };
-  
+
   const handleCreateMovie = async ({ title, description, runtimeMins }) => {
-    
-  }
+    const token = localStorage.getItem("token");
+
+    if (!token) {
+      console.error("No token found in local storage. Please log in.");
```

Comment:
This would be another good candidate for displaying an error to the user. Users _very rarely_ look at the console for errors.